### DISCUSSION TITLE
On definitions of coordinate systems

### DISCUSSION THREAD

**Discussion by GCChen97** [mitsuba-renderer/mitsuba-blender#47] [Q&A] [answered]
Hi,

Thanks for the convenient tool. I notice that the definitions of coordinate systems in mitsuba and blender are different when render normal map with AOV. It seems that the y axis in blender is z axis in mitsuba. So, what happens? Is SE3 transformation required when `import`/`export` scene.xml?

**Comment by GCChen97** [comment]
I also noticed that the ply meshes imported by the addon `mitsuba-blender` are -90 degree rotated around x axis comparing to those imported by `blender`. But obj meshes are consistently imported by both `mitsuba-blender` and `blender`.

**Reply by ros-dorian** [reply]
Hello, thank you for the comment.

Mitsuba and Blender have different *Forward* and *Up* vectors defining their coordinate systems. This means that if you are exporting a mesh from Blender or importing it from a Mitsuba scene (without using the add-on), you need to specify the correct values in the export/import dialog which are *-Z Forward* and *Y Up*.
Note that the add-on is doing this automatically when exporting/importing so it is always recommended to use it whenever possible.

Let me know if that answers your question.

Cheers!

**Reply by GCChen97** [reply]
Hi, thanks for the help. I further did some search and finally realized that _Y UP_  means the _XOZ_ plane is the "floor" and _Z UP_ means the _XOY_ plane is the "floor", which are not the definitions of camera frame. 

So I have another question: what's the definition of camera coordinate system in `blender` and `mitsuba`?  I guess _Y UP_ also means "screen top==Y up" in camera frame. If so, are _"+Z-forward"  and "Y-up" and "X-left"_ the definitions of camera frame in `mitsuba`, and _"-Z-forward"  and "Y-up" and "X-right"_ in `blender`?

**Reply by GCChen97** [reply]
I also noticed some \pi rotation around X/Y in the codes of `mitsuba-blender`. What are these codes doing?

**Reply by ros-dorian** [reply]
As you correctly suggested in the discussion over at [mitsuba-renderer/mitsuba3](https://github.com/mitsuba-renderer/mitsuba3/discussions/309#discussioncomment-3838187), the camera's forward direction is rotated 180 degrees around the up direction compared to the world frame. As Blender does not have the same definition, this code corrects that.

**Reply by GCChen97** [reply]
Thanks for the help! I am not familiar with CG and was confused about the differences between some common definitions in CV and CG.

**Comment by GCChen97** [comment, marked as answer]
Here are the definitions:
In mitsuba, the camera frame is "screen left==x, screen top==y and forward==z", the world frame is "Y up and -Z forward" (the "floor" is the XOZ plane).
In blender, the camera frame is "screen right==x, screen top==y and forward==-z", the world frame is "Z up and Y forward" (the "floor" is the XOY plane).

Thank @ros-dorian again!
